packages/sign_in_with_apple/android/src/main/kotlin/com/aboutyou/dart_packages/sign_in_with_apple/SignInWithApplePlugin.kt

```diff
@@ -0,0 +1,148 @@
+    }
+  }
+
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+    when (call.method) {
+      "isAvailable" -> result.success(true)
+      "performAuthorizationRequest" -> {
+        val _activity = activity
+
+        if (_activity == null) {
+          result.error("MISSING_ACTIVITY", "Plugin is not attached to an activity", call.arguments)
+          return
+        }
+
+        val url: String? = call.argument("url")
+
+        if (url == null) {
+          result.error("MISSING_ARG", "Missing 'url' argument", call.arguments)
+          return
+        }
+
+        SignInWithApplePlugin.lastAuthorizationRequestResult?.error("NEW_REQUEST", "A new request came in while this was still pending. The previous request (this one) was then cancelled.", null)
+        if (SignInWithApplePlugin.triggerMainActivityToHideChromeCustomTab != null) {
+          SignInWithApplePlugin.triggerMainActivityToHideChromeCustomTab!!()
+        }
+
+        SignInWithApplePlugin.lastAuthorizationRequestResult = result
+        SignInWithApplePlugin.triggerMainActivityToHideChromeCustomTab = {
+          val notificationIntent = _activity.getPackageManager().getLaunchIntentForPackage(_activity.getPackageName());
+          notificationIntent.setPackage(null)
+          notificationIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED);
+          _activity.startActivity(notificationIntent)
+        }
+
+        val builder = CustomTabsIntent.Builder();
+        val customTabsIntent = builder.build();
+        customTabsIntent.intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+        customTabsIntent.launchUrl(_activity, Uri.parse(url));
+      }
+      else -> {
+        result.notImplemented()
+      }
+    }
+  }
+
+  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+    activity = binding.activity
+  }
+
+  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+    onAttachedToActivity(binding)
+  }
+
+  override fun onDetachedFromActivityForConfigChanges() {
+    onDetachedFromActivity()
+  }
+
+  override fun onDetachedFromActivity() {
+    activity = null
+  }
+}
+
+/**
+ * Activity which is used when the web-based authentication flow links back to the app
+ *
+ * DO NOT rename this or it's package name as it's configured in the consumer's `AndroidManifest.xml`
+ *
+ */
+public class SignInWithAppleCallback: Activity {
+  constructor() : super()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    val lastAuthorizationRequestResult = SignInWithApplePlugin.lastAuthorizationRequestResult
+    if (lastAuthorizationRequestResult != null) {
+      lastAuthorizationRequestResult.success(intent?.data?.toString())
+      SignInWithApplePlugin.lastAuthorizationRequestResult = null
+    } else {
+      SignInWithApplePlugin.triggerMainActivityToHideChromeCustomTab = null
```

Comment:
Why are we setting `triggerMainActivityToHideChromeCustomTab ` here to `null`?

Reply:
While in practice they could only be `null` together, this seems still out of place

Reply:
Just as a clean up before we return. Most certainly it's already `null` as this point (with the current code structure at leat)